src/views/workflow-history/hooks/use-workflow-history-fetcher.ts

```diff
@@ -40,21 +46,25 @@ export default function useWorkflowHistoryFetcher(
 
   useEffect(() => {
     if (!fetcherRef.current) return;
-
     const unsubscribe = fetcherRef.current.onChange((state) => {
       const pagesCount = state.data?.pages?.length || 0;
+      // If the pages count is greater than the last flattened pages count, then we need to flatten the pages and call the onEventsChange callback
+      // Depending on ref variable instead of historyQueue is because historyQueue is throttled.
+      if (pagesCount > lastFlattenedPagesCountRef.current) {
+        lastFlattenedPagesCountRef.current = pagesCount;
+        onEventsChange(
+          state.data?.pages?.flatMap((page) => page.history?.events || []) || []
+        );
+      }
```

Comment:
I must be missing something.

My understanding:
- When `onChange` fires
- Then emit `onEventsChange` with fresh data

I would have expected:
- Within `onChange`
- Call `onEventsChange` with `state.data.pages`

I'm a bit confused why we don't pass `state.data.pages` directly (bypassing the flattening step)? 

The hook, as currently implemented, knows too much about the fetcher internals.